include/dynamatic/Transforms/Speculation/SpeculationPlacement.h

```diff
@@ -0,0 +1,108 @@
+//===- SpeculationPlacement.h - Speculation units placement -----*- C++ -*-===//
+//
+// Dynamatic is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the data structures needed for speculation algorithms
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef DYNAMATIC_TRANSFORMS_SPECULATION_PLACEMENT_H
+#define DYNAMATIC_TRANSFORMS_SPECULATION_PLACEMENT_H
+
+#include "circt/Dialect/Handshake/HandshakeOps.h"
+#include "dynamatic/Support/DynamaticPass.h"
+#include "dynamatic/Support/LLVM.h"
+#include "dynamatic/Support/Logging.h"
+#include "dynamatic/Support/TimingModels.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Pass/Pass.h"
+#include "llvm/ADT/DenseSet.h"
+#include "llvm/ADT/Hashing.h"
+#include <map>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+namespace dynamatic {
+namespace speculation {
+
+struct PlacementNames {
+  StringRef srcOpName;
+  StringRef dstOpName;
+};
+
+struct OpPlacement {
```

Comment:
With the JSON format change I propose, this struct can probably just be replaced by an `mlir::OpOperand&` (a reference to an operation's operand i.e., a value use).

Reply:
For convenience, now I am using a struct that is storing the `opName` as a string and `opIdx` as an unsigned integer.